src/server/worker/common/common.go

```diff
@@ -23,16 +23,12 @@ func IsDone(ctx context.Context) bool {
 
 // DatumID computes the ID of a datum.
 func DatumID(inputs []*Input) string {
-	// TODO: This is a stopgap solution that needs to be addressed before 2.0 GA.
-	// Datum IDs need more design work.
-	if inputs[0].GroupBy != "" {
-		return inputs[0].GroupBy
-	}
-	var files []string
+	hash := pfs.NewHash()
 	for _, input := range inputs {
```

Comment:
Are the inputs ordered?

Reply:
Yeah, they are ordered with respect to how they are defined in the pipeline spec.